src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -804,7 +822,31 @@ private void RefreshAvailableUpdatesCount()
 
                 _topPanel._labelUpgradeAvailable.Count = Model.CachedUpdates.Packages.Count;
             })
-            .FileAndForget(TelemetryUtility.CreateFileAndForgetEventName(nameof(PackageManagerControl), nameof(RefreshAvailableUpdatesCount)));
+            .FileAndForget(TelemetryUtility.CreateFileAndForgetEventName(nameof(PackageManagerControl), nameof(RefreshInstalledAndUpdatesTabs)));
+        }
+
+        private static async Task<int> GetInstalledDeprecatedPackagesCountAsync(PackageLoadContext loadContext, IPackageMetadataProvider metadataProvider, CancellationToken token)
+        {
+            var installedPackages = await loadContext.GetInstalledPackagesAsync();
```

Comment:
Why is this not explicitly on a background thread?

Reply:
It was unclear to me if it should be run on the main thread or not. At first, I had seen it called on the UI thread in several places so that's why I had it here. Upon further investigation, it appears that it is also called off the UI thread in some places. Any thoughts?

Reply:
Discussed offline--it does not need to be on the UI thread.